dotbot/controller.py

```diff
@@ -147,25 +150,28 @@ def on_byte_received(byte):
             """Callback called on byte received."""
             event_loop.call_soon_threadsafe(queue.put_nowait, byte)
 
-        async def _wait_for_handshake(queue):
-            """Waits for handshake reply and checks it."""
-            try:
-                byte = await queue.get()
-            except asyncio.exceptions.CancelledError as exc:
-                raise SerialInterfaceException("Handshake timeout") from exc
-            if int.from_bytes(byte, byteorder="little") != PROTOCOL_VERSION:
-                raise SerialInterfaceException("Handshake failed")
-
-        self.serial = SerialInterface(
-            self.settings.port, self.settings.baudrate, on_byte_received
-        )
+        if self.settings.port == "sailbot-simulator":
+            self.serial = SailBotSimSerialInterface(on_byte_received)
+        else:
+            async def _wait_for_handshake(queue):
```

Comment:
in #120, IIRC, all this indent out of large blocks is not required. The handshake can also be handled by the simulator.